### PR TITLE
Add Signature type

### LINKED ISSUE #11: Get ItemFn's fields match those of syn::ItemFn
`syn::ItemFn` is defined like
```rust
pub struct ItemFn {
    pub attrs: Vec<Attribute>,
    pub vis: Visibility,
    pub sig: Signature,
    pub block: Box<Block>,
}
```
But `syn-mid::ItemFn` is like 
```rust
pub struct ItemFn {
    pub attrs: Vec<Attribute>,
    pub vis: Visibility,
    pub constness: Option<Const>,
    pub asyncness: Option<Async>,
    pub unsafety: Option<Unsafe>,
    pub abi: Option<Abi>,
    pub fn_token: Fn,
    pub ident: Ident,
    pub generics: Generics,
    pub paren_token: Paren,
    pub inputs: Punctuated<FnArg, Comma>,
    pub output: ReturnType,
    pub block: Block,
}
```

I think it would be neat to have them in sync, collapsing most of it to `syn::Signature`; so people could move from `syn {feature = "full}` to `syn-mid` by simple `use syn::ItemFn` => `use syn-mid::ItemFn` replacement.

### PR DESCRIPTION
**Pull request by taiki-e**
Closes #11

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -71,6 +72,15 @@ ast_struct! {
     pub struct ItemFn {
         pub attrs: Vec<Attribute>,
         pub vis: Visibility,
+        pub sig: Signature,
+        pub block: Block,
```
Comment (CreepySkeleton) [inline src/lib.rs]: Could you please use `Box<Block>` here? This is the way it's done in syn.

File: src/lib.rs [hunk 2]
```diff
@@ -80,19 +90,20 @@ ast_struct! {
         pub generics: Generics,
         pub paren_token: token::Paren,
         pub inputs: Punctuated<FnArg, Token![,]>,
+        pub variadic: Option<Variadic>,
         pub output: ReturnType,
-        pub block: Block,
     }
 }
```
Comment (CreepySkeleton) [inline src/lib.rs:96]: I suggest you get rid of this `Signature` and simply reexport it from `syn`.

**Comment by CreepySkeleton** [comment]
@taiki-e Could you please explain why you need to redefine a lot of `struct`/`enum`s in `pat.rs` and `arg.rs`? In my shallow view they can be simply reexported from `syn` too.

File: src/lib.rs (hunk 2 above)
Comment (taiki-e) [reply src/lib.rs:96]: IIRC, [`syn::Signature`](https://docs.rs/syn/1.0.11/syn/struct.Signature.html) requires `full` feature of `syn`.

**Comment by taiki-e** [comment]
> Could you please explain why you need to redefine a lot of `struct`/`enum`s in `pat.rs` and `arg.rs`? In my shallow view they can be simply reexported from `syn` too.

IIRC, those types require `full` feature of `syn`.

EDIT: one of the aims of this crate is to avoid to depend on `full` feature of `syn`.

File: src/lib.rs (hunk 2 above)
Comment (CreepySkeleton) [reply src/lib.rs:96]: Yeah, that's quite sad...

File: src/lib.rs (hunk 1 above)
Comment (taiki-e) [reply src/lib.rs]: done.

**Comment by taiki-e** [comment]
Published in 0.5.0.

**Pull request merged**